cli/src/main.rs

```diff
@@ -1,19 +1,561 @@
+            true => start_background(http_port, rpc_port).await,
+        },
+        Err(e) => error!("Listening Error: {:?}", e),
+    }
+
+    Ok(())
+}
+
+async fn start_background(http_port: &u16, rpc_port: &u16) {
+    let cargo = env::var("CARGO").unwrap_or_else(|_| "cargo".to_string());
+    let _ = Command::new(cargo)
+        .env("RUST_LOG", "off")
+        .current_dir(project_root())
+        .args(&[
+            "run",
+            "--quiet",
+            "--package=development_server",
+            "--",
+            format!("--http-port={}", http_port).as_str(),
+            format!("--rpc-port={}", rpc_port).as_str(),
+        ])
+        .stdout(Stdio::null())
+        .spawn()
+        .expect("Unable to spawn development_server");
+}
+
+async fn stop(domain: &str, rpc_port: &u16) -> Result<(), DynError> {
+    if let Err(e) = try_stop(domain, rpc_port).await {
+        error!("{}", e);
+
+        std::process::exit(-1);
+    }
+
+    Ok(())
+}
+
+async fn try_stop(domain: &str, rpc_port: &u16) -> Result<(), DynError> {
+    info!("Stopping development server");
+    let address = format!("http://{}:{}", domain, rpc_port);
+    let client = DevelopmentClient::connect(address.clone()).await;
+    match client {
+        Ok(mut client) => {
+            let request = tonic::Request::new(Empty {});
+            let response = client.stop_server(request).await?;
+
+            match response.into_inner() {
+                Empty {} => warn!("Stopped development server"),
+            }
+        }
+        Err(err) => {
+            return match err.to_string().as_str() {
+                "transport error" => {
+                    warn!("Server not yet started");
+                    return Ok(());
+                }
+                err => *Box::new(Err(err.into())),
+            };
+        }
+    };
+
+    Ok(())
+}
+
+/*
+ * Usage:
+ *
+ * cargo run deploy --component=game
+ *
+ * This will take the file "./components/game.wasm" and deploy it.
+ */
+async fn deploy(
+    domain: &String,
+    http_port: &u16,
+    rpc_port: &u16,
+    component: &String,
+) -> Result<(), DynError> {
+    if let Err(e) = try_deploy(domain, http_port, rpc_port, component).await {
+        error!("{}", e);
+
+        std::process::exit(-1);
+    }
+
+    Ok(())
+}
+
+async fn try_deploy(
+    domain: &String,
+    http_port: &u16,
+    rpc_port: &u16,
+    component: &String,
+) -> Result<(), DynError> {
+    let server_state = poll_server_state(domain, rpc_port, &false).await;
+    if server_state
+        .as_ref()
+        .is_ok_and(|s| s == &ServerState::NotStarted)
+    {
+        let _ = start(domain, http_port, rpc_port, &false, &true).await;
+        let _ = poll_server_state(domain, rpc_port, &true).await;
+    };
+
+    if server_state.is_ok() {
+        let address = format!("http://{}:{}", domain, rpc_port);
+        let path = format!("./components/{}.wasm", component);
```

Comment:
This is perfect for now. 

Quick thought for the future. 

It prob makes sense to declare these in some `project.json / toml / yaml` in a follow up PR. Mind putting some thought here and starting a thread in the cli discord channel with an approach?

Reply:
Will do!

Think we can use [cargo [env]](https://doc.rust-lang.org/cargo/reference/config.html), but I'll do some more research before proposing something on #CLI